### ISSUE TITLE
 CMS, design system, components [SCREENREADER, KEYBOARD]: Aside SHOULD not be contained in another landmark

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#6379]
#  [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

```diff
! Team affected: VSA Public Websites; Project: AMA ph1
```

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The `<aside>` **SHOULD** not be contained in another landmark. 

Inside the `<main class="va-l-detail-page">` there is `<aside class="va-nav-linkslist va-nav-linkslist--related">` which is not allowed. This **best practice** rule ensures that `aside` elements or elements with `role=complementary` are not an internal part of another ARIA landmark. Nesting landmarks create confusing document structures. This rule is similar to existing rules that test this same ARIA requirement for the banner, contentinfo, and main roles.

### Why it Matters

Embedding an `<aside>` element in another landmark may disable screen reader functionality allowing users to navigate through complementary content. Complementary content is ancillary content to the main theme of a document or page. Screen reader users have the option to skip over complementary content when it appears at the top level of the accessibility API. 

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screenreader user, I want to navigate the content by landmarks and not miss any of the landmark content.

## Environment

* Operating System: any
* Browser: any
* Screenreading device: any
* Server destination: production

## Steps to Recreate

1. Enter `https://www.va.gov/decision-reviews/` in browser
2. Start screenreading device listed in Environment
3. Navigate by landmarks
4. Verify that the aside is missed
5. Alternately, run an axe-coconut browser scan

## Possible Fixes (optional)

Reconsider the structural relationship of the `<main>` and `<aside>` elements. 

The page template should be revised. Consider making the `<main>` element a `<section>`, for example, and using the `<main>` element on the main focus of the page. Landmarks should not be inside other landmarks.

## WCAG or Vendor Guidance (optional)

* [Complementary landmarks and/or asides are top level](https://dequeuniversity.com/rules/axe/3.5/landmark-complementary-is-top-level?application=coconut)
* [W3C ARIA Landmarks](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/complementary.html)

## Screenshots

![aside-inside-main-landmark-01](https://user-images.githubusercontent.com/57469/76537900-d0b86f80-6454-11ea-82c7-74932a356f79.png)

![aside-inside-main-landmark-02](https://user-images.githubusercontent.com/57469/76537805-b1b9dd80-6454-11ea-9da3-d766a29fb921.png)

**Comment by RLHecht** [comment]
@jenstrickland I'm not sure I know what it means to have an "aside" embedded. Could you do a screen shot of the issue here? Thanks!

**Comment by jenstrickland** [comment]
@RLHecht I added screenshots to the issue. It's a more complicated issue, that is likely to need developer attention. The structure of the page HTML is the issue. I also added some additional details in the Potential fix section.

**Comment by ncksllvn** [comment]
CC: @Mottie noticed there's an `<aside>` in `disability-benefits/996/components/IntroductionPage.jsx`

**Comment by kelsonic** [comment]
@jenstrickland Do we need an `<aside />` here? Could we just convert it to a `<div />`? 🙂

**Comment by ncksllvn** [comment]
Yeah what @kelsonic  said. Or is the better fix to restructure the page to move the `aside` out of the `main`?

**Comment by jenstrickland** [comment]
Converting the `<aside>` to a `<div>` is problematic. Other instances on VA.gov where the aside is coded as a div create issues for screen reader users, who have no context to distinguish between the aside content and everything else. It's a wall of words. To use a coder saying, "if everything's !important, then nothing is important." Having the `<main>` element wrap the entirety of the page content wipes out semantics for those who can't see the visual hierarchy. The `<main>` element is used as a landmark on the page, for screen reader users to locate the highest priority content on the page.

In the ticket section for Possible fixes, making the current `<main>` element a `<section>` and determining what content is the `<main>` content would be the preferred solution.

**Comment by ncksllvn** [comment]
> In the ticket section for Possible fixes, making the current <main> element a <section> and determining what content is the <main> content would be the preferred solution.

Hm, okay. That's more difficult than what we were thinking. Since this is a global issue occurring across most pages of the website, I don't think we should tie this to the AMA work.

**Comment by jenstrickland** [comment]
💯 agree. The templates your team is responsible is one approach to address it. Things are cloudy since we have two teams owning Drupal templates, and then other page templates. Where do we bring this up? @1Copenut  - should we have changes started within individual teams and the templates they own, or does this need to be addressed globally all at once?

**Comment by kelsonic** [comment]
I think the title of this ticket description is misleading. According to Accessibility's rubric, this is not a "launch blocker".

https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2

> **These issues will not delay your product launch.**

**Comment by jenstrickland** [comment]
@kelsonic The day I wrote this ticket 508-defect-2 was categorized as a launchblocker. @1Copenut changed it yesterday.

**Comment by 1Copenut** [comment]
For general awareness, the [severity rubric](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md) now identifies severity 0/1 issues as launch blockers. We've tried to be very specific on those items--what they look like, how they present, how they can be remediated. It'll continue to evolve and adapt, but we will make sure teams are aware if the list of launch blockers is changing.

**Comment by ncksllvn** [comment]
Launch blocker or not, this is an issue across hundreds of pages, rather than the specific subset that the AMA launch includes. I think it would make sense for it to be categorized or related to an Epic for site-wide/global issues.

**Comment by kelsonic** [comment]
> I think it would make sense for it to be categorized or related to an Epic for site-wide/global issues.

@ncksllvn @MickinSahni @1Copenut @jenstrickland What is the process for reporting and prioritizing to the correct team a global/site-wide accessibility issue? Would it be helpful if every sprint or so there were an Accessibility PM that could assign site-wide accessibility issues to the relevant team?

Just brainstorming on how to better this process -- we do really want to help with global accessibility issues, we really don't want new features to be blocked by systemic accessibility issues that should be address separately, and we know the Accessibility team really wants progress to be made on systemic accessibility issues. We should brainstorm a potential process that best facilitates this imo 🙂

**Comment by jenstrickland** [comment]
Recently, @1Copenut and I met with @kevwalsh to find out how changes get made to templates. He said individual teams own certain templates. Trevor and I understood your team owns the templates you're working on. They are separate from other pages, afaik. 

Are the templates pulling from global/site-wide stylesheets/components?

**Comment by ncksllvn** [comment]
They generally start at [layouts](https://github.com/department-of-veterans-affairs/vets-website/tree/master/src/site/layouts) but can include most of the sibling directories. I would consider trying to be agnostic to the underlying templating (unless you want to lend a hand on the implementations) and identify the general scope of a particular issue - is this site-wide, unique to just this page, just this area, etc. Might be easier said than done though.

**Comment by jenstrickland** [comment]
- At this time, [this particular issue comes up in axe browser scans](https://dequeuniversity.com/rules/axe/3.5/landmark-complementary-is-top-level?application=coconut) because someone semantically coded the sidebar content as an aside. 
- In pages that are not Drupal templates, these areas currently are coded as divs, and that's hiding the error. It doesn't provide context for assistive tech. 
- If you use the axe Chrome extension on template pages and compare with non-template pages, you'll see the difference. However, that `<div>` is not solving the user problem.
- Completely understand the scale of the complexity. If it can be fixed in the templates, that would be a start to improving the assistive tech UX. If it's too complex for now and would benefit from scale planning, I get that, too.

**Comment by MickinSahni** [comment]
Hi Trevor, should this pass to you as it's a global issue? @1Copenut

**Comment by 1Copenut** [comment]
@MickinSahni Yeah, I'll take it as a global issue and follow up with the triage team for next steps.

**Comment by MickinSahni** [comment]
Sweet. Dropping the `vsa-public-websites` label and AMA Public Websites Epic. Thanks.

**Comment by ncksllvn** [comment]
I'm gonna take a brief look into this. It seems like it's only used by one Drupal template - the  ordinary `page` layout.

**Comment by ncksllvn** [comment]
I think we can do this pretty easily for this element by just dropping moving the aside the rest of the layout and redefining a grid markup. It would look like -

```
<div  id="content">
     <main>
         <sidebar/>
          <article/>
      </main>
      <related-links>
</div>
```

There's a bunch of grid-classes wrapping elements inside the `<main>` element, and that can't be refactored for the `page` layout without also touching a bunch of other templates. The grid system being used there is also kinda outdated - it's using the USWDS grid classes instead of the VA flexbo-based grid classes.

So I think we can address this particular bug pretty easily, but the more high-level fix would be to define the most efficient HTML page layout that should be applied globally, then implement that across all page templates while also updating the grid classes.

**Comment by ncksllvn** [comment]
@jenstrickland Here's the template I'm referring to - https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/layouts/page.drupal.liquid. Check out the sibling files to see what I mean about the grid classes and how they affect the landmark elements.

**Comment by jenstrickland** [comment]
I realize this is a temporary patch, to remove the `<aside>` element. In the actual fix, the `<main>` and `<aside>` elements wouldn't be contained inside each other. What is the `<sidebar/>` reference? Is the recommendation to revert to the `<div class="left-side-nav-title">` or some other non-landmark code, for now?

**Comment by ncksllvn** [comment]
> What is the `<sidebar/>` reference? Is the recommendation to revert to the `<div class="left-side-nav-title">` or some other non-landmark code, for now?

I only included the sidebar there to be descriptive in that example while I was trying to describe how the grid classes make restructuring tricky. The sidebar seems like it should be a landmark of its own though.

**Comment by jenstrickland** [comment]
Gotcha! 

Yes, the sidebar would ideally be in an `<aside>` and outside of the `<main>`. As it is extra complicated, the quick fix would be to make it like the rest of the side, nullify for now into a `<div>`, until Platform addresses the sitewide issue. 

Since there isn't a sidebar on this page, is it possible to remove it? Are there other pages that use the template where there is content in the aside?

**Comment by ncksllvn** [comment]
> Since there isn't a sidebar on this page, is it possible to remove it? Are there other pages that use the template where there is content in the aside?

Oh I was actually just speaking in general about how the layout for this page is used but not this page exactly. The page that this issue specifically refers to doesn't have a sidebar so it doesn't matter for that. 

The content in the `<aside>` is always "Related links" that renders in a gray box at the bottom of the screen.  For example, on this page - https://staging.va.gov/health-care/eligibility/active-duty/ - the `Other helpful information` box is an `<aside>`.

**Comment by jenstrickland** [comment]
Oooh, that Related links and Other helpful information would be fine in a div that has [role="complementary"](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/complementary.html), yet would still need to not be contained by the main element. The primary trouble is that the main element was improperly used across the pages. :( I understand how hard this one is to fix. 

For this template, could the aside be changed into an article, for now?

**Comment by ncksllvn** [comment]
It can be changed but there's already an `article` on that page

**Comment by jenstrickland** [comment]
A page may have unlimited `<article>`s. They aren't landmark level.

**Comment by SKasimow** [comment]
@brianalloyd Thanks Brian! I wasn't sure where this belongs so will circle back with you after you meet with team.

**Comment by brianalloyd** [comment]
Gist: Calling for a change to side element on CV page would be calling for change across the site and not unique to just this page. This would result in a great deal of template changes, also would result in a high risk venture based on the widespread nature of the templates site wide. This would need a dedicated developer focus time to work this issue (Plan a full sprint). @SKasimow @youngfreezyVA

**Comment by laflannery** [comment]
No longer relevant, there is no nested aside on the page.  Will close the issue in 1 week unless told otherwise
